egs/wsj/s5/steps/nnet3/libs/xconfig_layers.py

```diff
@@ -0,0 +1,679 @@
+            cur_node = '{0}.affine'.format(self.name)
+            if presoftmax_scale_file != '' and config_name == 'all':
+                # don't use the presoftmax-scale in 'ref.config' since that file won't exist at the
+                # time we evaluate it.  (ref.config is used to find the left/right context).
+                line = ('component name={0}.fixed-scale type=FixedScaleComponent scales={1}'.format(
+                        self.name, presoftmax_scale_file))
+                ans.append((config_name, line))
+                line = ('component-node name={0}.fixed-scale component={0}.fixed-scale input={1}'.format(
+                        self.name, cur_node))
+                ans.append((config_name, line))
+                cur_node = '{0}.fixed-scale'.format(self.name)
+            if include_log_softmax:
+                line = ('component name={0}.log-softmax type=LogSoftmaxComponent dim={1}'.format(
+                        self.name, output_dim))
+                ans.append((config_name, line))
+                line = ('component-node name={0}.log-softmax component={0}.log-softmax input={1}'.format(
+                        self.name, cur_node))
+                ans.append((config_name, line))
+                cur_node = '{0}.log-softmax'.format(self.name)
+            line = ('output-node name={0} input={0}.log-softmax'.format(self.name, cur_node))
+            ans.append((config_name, line))
+        return ans
+
+
+# This class is for lines like
+#  'relu-renorm-layer name=layer1 dim=1024 input=Append(-3,0,3)'
+# or:
+#  'sigmoid-layer name=layer1 dim=1024 input=Append(-3,0,3)'
+# Here, the name of the layer itself dictates the sequence of nonlinearities
+# that are applied; the name should contain some combination of 'relu', 'renorm',
+# 'sigmoid' and 'tanh', and these nonlinearities will be added after the
+# affine component.
+#
+# The dimension specified is the output dim; the input dim is worked out from the input descriptor.
+# This class supports only nonlinearity types that do not change the dimension; we can create
+# another layer type to enable the use p-norm and similar dimension-reducing nonlinearities.
+#
+# See other configuration values below.
+#
+# Parameters of the class, and their defaults:
+#   input='[-1]'             [Descriptor giving the input of the layer.]
+#   dim=-1                   [Output dimension of layer, e.g. 1024]
+#   self-repair-scale=1.0e-05  [Affects relu, sigmoid and tanh layers.]
+#
+# Configuration values that we might one day want to add here, but which we
+# don't yet have, include target-rms (affects 'renorm' component).
+class XconfigSimpleLayer(XconfigLayerBase):
+    def __init__(self, first_token, key_to_value, prev_names = None):
+        # Here we just list some likely combinations.. you can just add any
+        # combinations you want to use, to this list.
+        assert first_token in [ 'relu-layer', 'relu-renorm-layer', 'sigmoid-layer',
```

Comment:
these names were a bit confusing for me. I would like to change these to affine-relu-layer, affine-relu-renorm-layer ...